### DISCUSSION TITLE
Strategies to change database when user disconnect

### DISCUSSION THREAD

**Discussion by JSmithOner** [supabase/supabase#8204] [General]
I know I can use authentication state changes and maybe (I don't know about that) realtime state changes to know if a user disconnect but I would like to knw if anyone has strategies when user closes all open sessions or token expires completly, in order to perform operations on the database?

Many thanks in advance.

**Comment by w3b6x9** [comment]
@JSmithOner Supabase Realtime recently announced the Presence feature. You can use that to track a user across open sessions and perform some action when user has closed all their sessions.

**Reply by JSmithOner** [reply]
@w3b6x9 Thank you for your answer but imagine someone switches tab or close browser could you explain to me how I could perform an action ( I guess this would be done on backend side (database functions or Edge Functions)) or do you have any examples.Best.

**Reply by w3b6x9** [reply]
@JSmithOner the action would be left to you to handle and you can trigger whatever you like.

**Reply by JSmithOner** [reply]
@w3b6x9 Sorry if you misunderstood my concern was more like "is there a built-in functionnality to do so?" for example a gate between Presence and a database or Edge function trigger because if user closes tab or navigator trigger should be backend, if you use `invoke` or any database action on the client chances are the operation could not complete.

**Reply by Diwannist** [reply]
I also have the same question. How do you trigger a backend action on user disconnect?
